File: Definitions/Game_Definitions/classes/NetrunnerControlPanel.lua
---@meta
---@diagnostic disable

---@class NetrunnerControlPanel : BasicDistractionDevice
NetrunnerControlPanel = {}

---@return NetrunnerControlPanel
function NetrunnerControlPanel.new() return end

---@param props table
---@return NetrunnerControlPanel
function NetrunnerControlPanel.new(props) return end

---@param evt FactQuickHack
---@return Bool
function NetrunnerControlPanel:OnCreateFactQuickHack(evt) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function NetrunnerControlPanel:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function NetrunnerControlPanel:OnTakeControl(ri) return end

---@return NetrunnerControlPanelController
function NetrunnerControlPanel:GetController() return end

---@return NetrunnerControlPanelControllerPS
function NetrunnerControlPanel:GetDevicePS() return end

File: DecompiledGameScripts/cyberpunk/devices/basicDistractionDevice/netrunnerControlPanel.swift

public class NetrunnerControlPanel extends BasicDistractionDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as NetrunnerControlPanelController;
  }

  protected cb func OnCreateFactQuickHack(evt: ref<FactQuickHack>) -> Bool {
    let properties: ComputerQuickHackData = evt.GetFactProperties();
    if Equals(properties.operationType, EMathOperationType.Set) {
      SetFactValue(this.GetGame(), properties.factName, properties.factValue);
    } else {
      AddFact(this.GetGame(), properties.factName, properties.factValue);
    };
  }

  private const func GetController() -> ref<NetrunnerControlPanelController> {
    return this.m_controller as NetrunnerControlPanelController;
  }

  public const func GetDevicePS() -> ref<NetrunnerControlPanelControllerPS> {
    return this.GetController().GetPS();
  }
}
